test/functional/plugin/lsp_spec.lua

```diff
@@ -1378,6 +1378,33 @@ describe('LSP', function()
       eq(4, pos.col)
       eq('å', exec_lua[[return vim.fn.expand('<cword>')]])
     end)
+
+    it('pushes to tagstack without repeating items', function()
+      local stack
+      -- Jump to a location.
+      jump(location(0, 9, 0, 9))
+      stack = exec_lua[[return vim.fn.gettagstack()]]
+      eq(1, stack.length)
+      eq({target_bufnr, 1, 1, 0}, stack.items[1].from)
+
+      -- Jump to same location. This should not be added to the tagstack.
+      jump(location(0, 9, 0, 9))
+      stack = exec_lua[[return vim.fn.gettagstack()]]
+      eq(1, stack.length)
+
+      -- Jump to a different location.
+      jump(location(0, 10, 0, 9))
+      stack = exec_lua[[return vim.fn.gettagstack()]]
+      eq(2, stack.length)
+      eq({target_bufnr, 1, 10, 0}, stack.items[2].from)
+    end)
+
+    it('prevents pushing to tagstack when motionless', function()
+      -- Jump to the same location the cursor is parked.
+      jump(location(0, 0, 0, 0))
+      local stack = exec_lua[[return vim.fn.gettagstack()]]
+      eq(0, stack.length)
+    end)
```

Comment:
Do you think you could add a test where we jump to a different buffer? I'm not sure we're doing that here. That would be good anyway, since it seems easy for us to mess up in the future.

Reply:
Like jump from buffer 1, to buffer 2, to buffer 1, for example.

Or, jump from buffer 1 to buffer 2, but in the same cursor location. That might be a good test as well.

Reply:
Makes sense, more tests the better!

Reply:
Done, that is a test case now, you can find it by reading the comments I left in the tests. :smile_cat:

Reply:
So another test case I'd like to see is that you do something with `location` where you set a new `uri` to go to. 

You may have to extend: https://github.com/neovim/neovim/pull/12262/files#diff-2a259d96bc36492344afe33e6d418d83R1361-R1369 to allow for a new `uri` and to have it opened with text inside of it in this section:  https://github.com/neovim/neovim/pull/12262/files#diff-2a259d96bc36492344afe33e6d418d83R1352-R1359

So basically, add another `uri`, like `fake/uri_2` or similar, and then jump there and back.

Also, using something like using `<C-t>` would be pretty cool here to see see if it actually jumps back to where you were before.

Reply:
Good point, I'd love to organize the whole test.

> So basically, add another uri, like fake/uri_2 or similar, and then jump there and back.

I tried that but unfortunately the test threw an error saying it couldn't switch to another buffer. :crying_cat_face: 

Basically, inside a `test` context, I can only call `nvim_set_current_buf` once. If I try to call it again (so I can jump between buffers), it throws that error. I have no clue whether I'm doing something wrong or it's a limitation of the test itself.

> Also, using something like using <C-t> would be pretty cool here to see see if it actually jumps back to where you were before.

This makes total sense, we need to prove whether we're properly pushing items to the tagstack. I need two things to make this possible:
- Solve the issue I cited above
- Learn how to trigger `<C-t>` (is it via `execute normal!`?)

Reply:
Hmm, it could be because you didn't have another buffer loaded. That's why I think you need to load it in the `before_each` section. Does that make sense?

As for the trigger the key sequence, you can use `helpers.feed`, which should work fine.

Reply:
I tried loading the new buffer using the same code from the `before_each` hook just before the call to `nvim_set_current_buf` (inside the `test` section), but I had no luck. I think I have even loaded it in the `before_each` hook, but still no success.

I'll try it again later and if I have no success, I'll ask for help here or on IRC. Thanks again!